File: client-go/internel/lib/sysinfo/src/monitor/memory_info.rs

use serde::{Deserialize, Serialize};
use sysinfo::System;

use crate::utils::{byte_to_gb, two_decimals};

#[derive(Debug, Serialize, Deserialize)]
pub struct MemoryInfo {
    total_memory: f64,
    used_memory: f64,
    free_memory: f64,
    percent_memory: f64,

    total_swap: f64,
    used_swap: f64,
    free_swap: f64,
    percent_swap: f64,
}

impl MemoryInfo {
    pub fn new() -> MemoryInfo {
        let mut sys = System::new_all();
        sys.refresh_all();

        let total_memory = byte_to_gb(sys.total_memory());
        let used_memory = byte_to_gb(sys.used_memory());
        let free_memory = byte_to_gb(sys.free_memory());
        let percent_memory = two_decimals(used_memory / total_memory);

        let total_swap = byte_to_gb(sys.total_swap());
        let used_swap = byte_to_gb(sys.used_swap());
        let free_swap = byte_to_gb(sys.free_swap());
        let percent_swap = two_decimals(used_swap / total_swap);

        // 直接返回 MemoryInfo 实例
        MemoryInfo {
            total_memory,
            used_memory,
            free_memory,
            percent_memory,
            total_swap,
            used_swap,
            free_swap,
            percent_swap,
        }
    }

    pub fn to_json(&self) -> String {
        serde_json::to_string(&self).unwrap()
    }
}

File: src/main/java/cv/beriholic/beeyes/models/dto/system/MemoryInfo.java
package cv.beriholic.beeyes.models.dto.system;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serial;
import java.io.Serializable;

/**
 * 内存信息包括RAM和交换分区
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MemoryInfo implements Serializable {
    @Serial
    private static final long serialVersionUID = 4049866545799310455L;
    /**
     * 总内存（字节）
     */
    private Long total_memory;

    /**
     * 已使用内存（字节）
     */
    private Long used_memory;

    /**
     * 可用内存（字节）
     */
    private Long free_memory;

    /**
     * 总交换分区（字节）
     */
    private Long total_swap;

    /**
     * 已使用交换分区（字节）
     */
    private Long used_swap;

    /**
     * 可用交换分区（字节）
     */
    private Long free_swap;

    /**
     * 内存使用百分比
     */
    private Double percent_memory;

    /**
     * 交换分区使用百分比
     */
    private Double percent_swap;
}
